src/core/thread/network_data_local.cpp

```diff
@@ -326,7 +326,7 @@ otError Local::SendServerDataNotification(void)
 #if OPENTHREAD_FTD
 
     // Don't send this Server Data Notification if the device is going to upgrade to Router
-    if (Get<Mle::MleRouter>().IsRouterEligible() && (Get<Mle::MleRouter>().GetRole() < OT_DEVICE_ROLE_ROUTER) &&
+    if (Get<Mle::MleRouter>().IsRouterEligible() && !Get<Mle::MleRouter>().IsRouterOrLeader() &&
```

Comment:
Adding a note: 

This should be the only place in this PR where the change is more than just a renaming (and a logic change).
Here we change the relative comparison of `GetRol()` value with the constant and instead use the `IsRouterOrLeader()`.